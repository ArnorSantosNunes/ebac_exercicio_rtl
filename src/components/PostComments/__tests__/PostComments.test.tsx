import { fireEvent, render, screen } from '@testing-library/react'
import Post from '..'

describe('Teste para o componente PostComment', () => {
    test('Deve renderizar o componente corretamente', () => {
        render(<Post />)
        expect(screen.getByText('Comentar')).toBeInTheDocument()
    })
    test('Deve adicionar "estudar react" na lista', () => {
    render(<Post />);

    const textarea = screen.getByTestId('campo-tarefa');
    const submitButton = screen.getByTestId('btn-cadastrar');
    const commentList = screen.getByTestId('lista-completa');

    // Primeiro comentário
    fireEvent.change(textarea, { target: { value: 'Comentário 1' } });
    fireEvent.click(submitButton);

    // Segundo comentário
    fireEvent.change(textarea, { target: { value: 'Comentário 2' } });
    fireEvent.click(submitButton);

    const comments = screen.getAllByTestId('campo-lista');

    expect(comments).toHaveLength(2);
    expect(comments[0]).toHaveTextContent('Comentário 1');
    expect(comments[1]).toHaveTextContent('Comentário 2');

    // Snapshot do container completo
    expect(commentList).toMatchSnapshot();
});
})