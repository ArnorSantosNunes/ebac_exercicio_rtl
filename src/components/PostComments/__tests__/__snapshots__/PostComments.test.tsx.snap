// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste para o componente PostComment Deve adicionar "estudar react" na lista 1`] = `
<ul
  class="post-comments"
  data-testid="lista-completa"
>
  <li
    class="post-comment"
    data-testid="campo-lista"
  >
    <p
      class="post-comment-content"
    >
      Comentário 1
    </p>
  </li>
  <li
    class="post-comment"
    data-testid="campo-lista"
  >
    <p
      class="post-comment-content"
    >
      Comentário 2
    </p>
  </li>
</ul>
`;
